ngraph/core/src/op/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,175 @@
+    const auto& logits_pshape = get_input_partial_shape(0);
+    const auto& seq_len_pshape = get_input_partial_shape(1);
+    auto input_et = get_input_element_type(0);
+    const bool logits_is_static_rank = logits_pshape.rank().is_static();
+    const bool seq_len_is_static_rank = seq_len_pshape.rank().is_static();
+
+    // output dynamic rank tensor if all inputs are of dynamic rank
+    if (!logits_is_static_rank && !seq_len_is_static_rank)
+    {
+        set_output_type(0, input_et, PartialShape{Dimension::dynamic(), Dimension::dynamic()});
+    }
+
+    // check ranks of input tensors
+    if (logits_is_static_rank)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              logits_pshape.rank().get_length() == 3,
+                              "The rank of logits tensor must be equal to 3.");
+    }
+    if (seq_len_is_static_rank)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              seq_len_pshape.rank().get_length() == 1,
+                              "The rank of sequence len tensor must be equal to 1.");
+    }
+
+    // check optional input type: blank index
+    if (get_input_size() == 3)
+    {
+        const auto& blank_index_type = get_input_element_type(2);
+        NODE_VALIDATION_CHECK(this,
+                              blank_index_type.is_integral_number(),
+                              "The blank index type is expected to be an integer type. Got: ",
+                              blank_index_type);
+
+        const auto& blank_index_shape = get_input_partial_shape(2);
+        Shape blank_index_type_shape = blank_index_shape.to_shape();
+        NODE_VALIDATION_CHECK(
+            this,
+            blank_index_shape.is_dynamic() || ngraph::is_scalar(blank_index_type_shape) ||
+                (is_vector(blank_index_type_shape) && (blank_index_type_shape[0] == 1)),
+            "Expected 0D or 1D tensor for the 'blank_index' input. Got: ",
+            blank_index_type);
+    }
+
+    // validate input shapes and compute output shape
+    ngraph::Dimension batch_size = Dimension::dynamic();
+    ngraph::Dimension time_size = Dimension::dynamic();
+
+    if (logits_is_static_rank)
+    {
+        if (logits_pshape[0].is_static())
+        {
+            batch_size = logits_pshape[0];
+        }
+        if (logits_pshape[1].is_static())
+        {
+            time_size = logits_pshape[1];
+        }
+    }
+
+    if (seq_len_is_static_rank && seq_len_pshape[0].is_static())
+    {
+        if (batch_size != Dimension::dynamic())
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  seq_len_pshape[0] == batch_size,
+                                  "The first dimensions of input tensors must match.");
+        }
+        batch_size = seq_len_pshape[0];
+    }
+
+    if (logits_is_static_rank && seq_len_is_static_rank &&
+        (logits_pshape[0] == Dimension::dynamic() && seq_len_pshape[0] == Dimension::dynamic()))
```

Comment:
Why do you check that dimensions are dynamic? The `seq_len_pshape[0] & logits_pshape[0]` should work for static/mixed dimensions also.